lib/ansible/playbook/taggable.py

```diff
@@ -42,12 +42,63 @@ def _load_tags(self, attr, ds):
         else:
             raise AnsibleError('tags must be specified as a list', obj=ds)
 
+    def _post_validate_tags(self, attr, value, templar):
+        '''Tags are templatable and use extend=True, so this value
```

Comment:
not just tags, i would make this a generic function for templatable and appendable FieldAttirbutes (tags, environment, module_defaults and conditionals)

Reply:
though .. conditionals can get 'iffy' since `when: []` is kindof valid

Reply:
Good point